torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -71,13 +71,18 @@ class KernelIrScanner : private kir::IrVisitor {
     // Do we have any reductions?
     summary_.has_block_reductions =
         summary_.has_block_reductions || domain->hasBlockReduction();
-    summary_.has_grid_reductions =
-        summary_.has_grid_reductions || domain->hasGridReduction();
 
     // Do we have block broadcasts?
     summary_.has_block_broadcasts =
         summary_.has_block_broadcasts || domain->hasBlockBroadcast();
 
+    // TODO: Why definition() does not return kir::GridReduction?
+    if (domain->hasGridReduction()) {
+      if (tensor_index->definition()->isA<kir::ReductionOp>()) {
```

Comment:
@tlemo Here, when `tensor_index` is used to call `gridReduce`, it seems `tensor_index->definition()` gives `kir::ReductionOp` instead of `kir::GridReduction`. Any idea why?

Reply:
Could it be the consequence of the path below?

https://github.com/csarofeen/pytorch/blob/dbd15ededdcb1be9dbac717cb8bad10607356d1d/torch/csrc/jit/codegen/cuda/lower_index.cpp#L252-L255

Reply:
Thanks. That makes sense.